### PR TITLE
refactor(fortran): complete handler with module extraction and call graph

### LINKED ISSUE #827: refactor(fortran): complete Fortran handler -- modules, call graph, extract_function_name, obj%method() support
## Summary

Complete the Fortran language handler (`fortran.rs`) to reach feature parity with other language handlers. The grammar (tree-sitter-fortran 0.6.0, already the latest) exposes all required node types; the gap is entirely in our query strings and missing handler implementations.

## Motivation

- Current pain point: module extraction is disabled by a stale comment asserting `module_statement` has no `name` child -- it does in 0.6.0. `extract_function_name`, `find_receiver_type`, and `find_method_for_receiver` are all `None`, disabling call graph traversal for Fortran entirely.
- Triggered by: Fortran analysis requirement for the Halliburton/slalom project.
- Expected benefit: full `analyze_file`, `analyze_symbol`, and `analyze_directory` output for Fortran codebases -- modules, subroutines, functions, USE imports, and CALL graphs all populated correctly.

## Current State

File: `crates/aptu-coder-core/src/languages/fortran.rs`

```rust
// ELEMENT_QUERY: only captures subroutine and function nodes; modules omitted
// CALL_QUERY: captures subroutine_call and call_expression identifiers only (misses obj%method() pattern)
// extract_function_name: None  -- no name extraction from subroutine_statement / function_statement
// find_receiver_type: None     -- no module-scoped procedure detection
// find_method_for_receiver: None
// defuse_query: None
```

The comment in `ELEMENT_QUERY` reads:

> `module_statement` has no `name` field in the current grammar

This was true for an older grammar version. In 0.6.0, `module_statement` exposes `name` as a required child node (not a named field), confirmed by `src/node-types.json` in the grammar source and the upstream `tags.scm`:

```scheme
(module_statement (name) @name) @definition.module
```

File: `crates/aptu-coder-core/src/languages/mod.rs` (lines ~208-222)

```rust
"fortran" => Some(LanguageInfo {
    // ...
    extract_function_name: None,       // call graph broken
    find_method_for_receiver: None,    // OOP-style obj%method() not tracked
    find_receiver_type: None,          // module-scoped procedures invisible
    defuse_query: None,
    // ...
})
```

### 1. Fix `ELEMENT_QUERY` -- add module capture

```rust
pub const ELEMENT_QUERY: &str = r"
(subroutine
  (subroutine_statement) @function)

(function
  (function_statement) @function)

(module
  (module_statement (name) @name) @class)

(module
  (module_subprogram_part
    (subroutine (subroutine_statement) @function)))

(module
  (module_subprogram_part
    (function (function_statement) @function)))
";
```

### 3. Fix `CALL_QUERY` -- add `part_ref` for `obj%method()` OOP-style calls

```rust
// obj%method(args) pattern -- Fortran 2003+ derived type bound procedures
(part_ref
  (data_ref
    name: (identifier) @call))
```

### 4. Implement `extract_function_name`

Extract `name` from `subroutine_statement` or `function_statement` nodes. Both expose `name` as a named field in 0.6.0.

```rust
pub fn extract_function_name(node: &Node, source: &str, _lang: &str) -> Option<String> {
    match node.kind() {
        "subroutine_statement" | "function_statement" => {
            node.child_by_field_name("name")
                .and_then(|n| get_node_text(&n, source))
        }
        _ => None,
    }
}
```

### 5. Implement `find_receiver_type` and `find_method_for_receiver`

Walk parent chain to detect if the subroutine/function is inside a `module` body. Return module name as receiver type; return procedure name as method.

### 6. Wire all handlers in `mod.rs`

```rust
"fortran" => Some(LanguageInfo {
    // ...
    extract_function_name: Some(fortran::extract_function_name),
    find_method_for_receiver: Some(fortran::find_method_for_receiver),
    find_receiver_type: Some(fortran::find_receiver_type),
    // defuse_query: defer to follow-up issue
    // ...
})
```

### 7. Tests (AAA pattern, one happy path + one edge case per behavior)

- Module name extraction from `MODULE foo ... END MODULE foo`
- Subroutine name extraction from `SUBROUTINE bar(...)`
- Function name extraction from `FUNCTION baz(...)`
- USE import extraction from `USE some_module`
- CALL graph: `CALL sub(args)` and `obj%method(args)`
- Module-scoped subroutine: `find_receiver_type` returns module name
- Top-level subroutine: `find_receiver_type` returns `None`

## Constraints

- No new dependencies; tree-sitter-fortran 0.6.0 is already the latest and sufficient
- No changes to `LanguageInfo` struct shape
- No changes outside `fortran.rs` and the `mod.rs` fortran registration block
- All existing tests must continue to pass
- `defuse_query` is out of scope for this issue (deferred)

## Acceptance Criteria

- [ ] `analyze_file` on a Fortran file lists modules, subroutines, and functions with correct names
- [ ] `analyze_symbol` produces a populated call graph for Fortran subroutine/function calls
- [ ] `analyze_directory` correctly counts modules (as classes) and functions in Fortran files
- [ ] USE statement imports appear in `analyze_file` import section
- [ ] All new tests follow AAA pattern and pass: `cargo test --features lang-fortran`
- [ ] No clippy warnings: `cargo clippy -- -D warnings`
- [ ] Code formatted: `cargo fmt --check`
- [ ] Commit GPG signed and DCO signed-off

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Completes the Fortran language handler to reach feature parity with Kotlin, C#, and Java. The grammar (tree-sitter-fortran 0.6.0, already the latest) exposes all required node types; the entire gap was in our query strings and missing handler implementations.

Fixes a stale code comment that disabled module extraction by incorrectly asserting `module_statement` had no `name` child. In 0.6.0, `name` is a required child (not a named field -- the original assumption was also wrong on the access method).

## Changes

- `crates/aptu-coder-core/src/languages/fortran.rs` -- fixed ELEMENT_QUERY (module capture via child iteration, CONTAINS sections via `internal_procedures`), fixed CALL_QUERY (added `derived_type_member_expression` for `obj%method()` calls), implemented `extract_function_name`, `find_receiver_type`, `find_method_for_receiver`, added 16 AAA-pattern tests
- `crates/aptu-coder-core/src/languages/mod.rs` -- wired all three handlers in the fortran `LanguageInfo` registration block (4-line change)

## Key implementation details

`module_statement.name` is a required **child**, not a named field. `child_by_field_name("name")` silently returns `None` -- corrected to use `children()` iteration matching `kind == "name"`.

`subroutine_statement.name` and `function_statement.name` are named **fields** -- `child_by_field_name("name")` is correct for these.

The CONTAINS container node is `internal_procedures`, not `module_subprogram_part` (the issue doc had this wrong; discovered empirically during BUILD).

`derived_type_member_expression` is the grammar node for `obj%method()` Fortran 2003+ bound procedure calls. `part_ref` does not exist in this grammar.

`defuse_query` is deferred to a follow-up issue.

## Test plan

- [x] 16 new Fortran-specific tests pass (module extraction, subroutine/function name extraction, USE imports, direct CALL and OOP member calls, module-scoped vs top-level scoping, CONTAINS sections, empty module edge case)
- [x] 230 total tests pass (214 pre-existing + 16 new)
- [x] `cargo clippy -- -D warnings` clean
- [x] `cargo fmt --check` clean
- [x] Security scan: no findings
- [x] No new dependencies
- [x] Only 2 files modified: `fortran.rs` and `mod.rs`

Closes #827

### CODE REVIEW

File: crates/aptu-coder-core/src/languages/fortran.rs [hunk 1]
```diff
@@ -36,17 +53,103 @@ pub const IMPORT_QUERY: &str = r"
...
+pub fn find_receiver_type(node: &Node, source: &str) -> Option<String> {
+    if !matches!(node.kind(), "subroutine_statement" | "function_statement") {
+        return None;
+    }
+    let mut current = *node;
+    while let Some(parent) = current.parent() {
+        if parent.kind() == "module" {
+            // Find the module_statement child and then its name child.
+            let mut cursor = parent.walk();
+            for child in parent.children(&mut cursor) {
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder-core/src/languages/fortran.rs]: Extracted into a private `extract_module_name` helper that centralises the two-level child walk on `module_statement`. `find_receiver_type` now delegates to it in a single call. The helper is private (no `pub`) so it does not widen the public API.

File: crates/aptu-coder-core/src/languages/fortran.rs [hunk 2]
```diff
@@ -36,17 +53,111 @@ pub const IMPORT_QUERY: &str = r"
...
+        }
+        current = parent;
+    }
+    None
+}
+
+/// Find the method name for a subroutine/function defined inside a module.
+/// Returns the function/subroutine identifier if the node is enclosed by a module.
+#[must_use]
+pub fn find_method_for_receiver(
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder-core/src/languages/fortran.rs:118]: False positive from the review tool -- the function is complete. It terminates with `node.child_by_field_name("name").and_then(|n| get_node_text(&n, source))` after the `in_module` guard check. All 230 tests (including `test_find_method_for_receiver_in_module` and `test_find_method_for_receiver_top_level`) pass, confirming the implementation is correct and fully terminated.
